src/module/applications/sheets/character.mjs

```diff
@@ -325,12 +335,48 @@ export default class DrawSteelCharacterSheet extends DrawSteelActorSheet {
     if (item.type === "class") {
       const cls = this.document.system.class;
       if (cls && (cls.dsid !== item.dsid)) {
-        ui.notifications.error("DRAW_STEEL.ADVANCEMENT.WARNING.cannotAddNewClass", { localize: true });
+        ui.notifications.error("DRAW_STEEL.ADVANCEMENT.WARNING.cannotAddNewType", {
+          format: { type: game.i18n.localize(CONFIG.Item.typeLabels[item.type]) },
+        });
         return;
       }
       return this.document.system.advance({ levels: 1, item });
+    } else if (item.system instanceof AdvancementModel) {
+      // Other advancements
+      if (["ancestry", "career", "culture"].includes(item.type)) {
+        /** @type {DrawSteelItem} */
+        const existing = this.document.system[item.type];
+        if (existing) {
+          const replace = await ds.applications.api.DSDialog.confirm({
+            window: {
+              icon: "fa-solid fa-arrow-right-arrow-left",
+              title: game.i18n.format("DRAW_STEEL.ADVANCEMENT.ReplaceItem.title", { type: game.i18n.localize(CONFIG.Item.typeLabels[item.type]) }),
+            },
+            content: `<p>${game.i18n.format("DRAW_STEEL.ADVANCEMENT.ReplaceItem.content", { name: item.name })}</p>`,
+          });
+          // TODO: Undo previous item's advancements
+          if (replace) await existing.delete();
+          else return;
+        }
+      }
+      else if (item.type === "kit") {
+        const actorClass = this.document.system.class;
+        if (actorClass?.system.kits === 0) {
+          const message = game.i18n.format("DRAW_STEEL.Item.kit.NotAllowedByClass", { class: actorClass.name });
+          ui.notifications.error(message);
+          return false;
+        }
+        const swapKit = await item.system.kitSwapDialog(this.document);
+        if (swapKit === false) return false;
+      }
+      return item.system.applyAdvancements({ actor: this.document, levelStart: 1, levelEnd: this.document.system.level });
```

Comment:
For classes, we use `range` (a tuple of numbers). We should align there - I'm fine with `levelEnd` and `Start` but perhaps `range: { start, end }`?

Reply:
I think those names would make more intuitive sense.